avm/src/vm/opcode.rs

```diff
@@ -3076,18 +3117,86 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
   }
   io!(httplsn => fn(_args, hand_mem) {
     Box::pin(async move {
-      let port_num = Program::global().http_port;
-      let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
-      let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(http_listener)) });
-
-      let bind = Server::try_bind(&addr);
-      match bind {
-        Ok(server) => {
-          let server = server.serve(make_svc);
+      match &Program::global().http_config {
+        HttpType::HTTP(http) => {
+          let port_num = http.port;
+          let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
+          let make_svc = make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(http_listener)) });
+
+          let bind = Server::try_bind(&addr);
+          match bind {
+            Ok(server) => {
+              let server = server.serve(make_svc);
+              tokio::spawn(async move { server.await });
+              println!("HTTP server listening on port {}", port_num);
+            },
+            Err(ee) => eprintln!("HTTP server failed to listen on port {}: {}", port_num, ee),
+          }
+        },
+        HttpType::HTTPS(https) => {
+          let port_num = https.port;
+          let addr = SocketAddr::from(([0, 0, 0, 0], port_num));
+          let tls_cfg = {
+            let certs = pemfile::certs(
+              &mut BufReader::new(base64::decode(https.cert_b64.as_str()).unwrap().as_slice())
+            );
+            let certs = match certs {
+              Ok(certs) => certs,
+              Err(_) => panic!("Failed to load certificate"),
+            };
+            eprintln!("{:?}", certs);
```

Comment:
maybe we should say `using certs for https: {:?}`?

Reply:
Oh, this is a debug print line that I forgot to remove.